PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+    if ($opensslVersion -match "OpenSSL 3.") {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath -traditional >> $null 2>$null
+    } else {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath >> $null 2>$null
+    }
+
+    # Clean up
+    remove-item $cfgFilePath
+    remove-item $orginalKeyFilePath
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsrFiles
+
+Function Invoke-RequestSignedCertificateByMsca {
+    <#
+        .SYNOPSIS
+        Request CSR to be signed by a MSCA Certificate Authority.
+
+        .DESCRIPTION
+        The Invoke-RequestSignedCertificateByMsca cmdlet connects to MSCA web enrollment to sign a CSR.  
+        
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware"
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io.
+
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware" -getCArootCert
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io and also will retrieve 
+        the Certificate Authority's root chain certificate.  
+
+        .PARAMETER caFqdn
+        The FQDN of the Certificate Authorty web enrollment website.
+
+        .PARAMETER username
+        The username to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER password
+        The password to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER csrFilePath
+        The full file path for the certificate signing request file. 
+
+        .PARAMETER certDir
+        The directory path to store the certificate authority signed certificate file.
+
+        .PARAMETER certificateTemplate
+        The name of the certificate template to be used.  
+
+        .PARAMETER getCArootCert
+        The Certificate Authorty chain root certificate will be retrieved as well.   
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $caFqdn,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $username,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $password,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $csrFilePath,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certDir,            
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certificateTemplate,
+        [Parameter (Mandatory = $false)] [Switch] $getCArootCert
```

Comment:
```suggestion
    Param (
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$caFqdn,
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$csrFilePath,
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$certDir,            
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$certificateTemplate,
        [Parameter (Mandatory = $false)] [Switch]$getCArootCert
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53